GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MultithreadedReadWriteTests.cs

```diff
@@ -14,8 +14,48 @@ namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
     [Category(Categories.Mac.M1)]
     public class MultithreadedReadWriteTests : TestsWithEnlistmentPerFixture
     {
-        [TestCase]
-        public void CanReadUnhydratedFileInParallelWithoutTearing()
+        [TestCase, Order(1)]
+        public void CanReadVirtualFileInParallel()
+        {
+            // Note: This test MUST go first, or else it needs to ensure that it is reading a unique path compared to the
+            // other tests in this class. That applies to every directory in the path, as well as the leaf file name.
+            // Otherwise, this test loses most of its value because there will be no races occurring on creating the
+            // placeholder directories, enumerating them, and then creating a placeholder file and hydrating it.
+            
+            string fileName = Path.Combine("GVFS", "GVFS.FunctionalTests", "Tests", "LongRunningEnlistment", "GitMoveRenameTests.cs");
+            string virtualPath = this.Enlistment.GetVirtualPathTo(fileName);
+
+            Exception readException = null;
+
+            Thread[] threads = new Thread[32];
+            for (int i = 0; i < threads.Length; ++i)
+            {
+                int myIndex = i;
+                threads[i] = new Thread(() =>
+                {
+                    try
+                    {
+                        FileSystemRunner.DefaultRunner.ReadAllText(virtualPath).ShouldBeNonEmpty();

```

Comment:
And that implies a need to make this class create an enlistment per test case, since the test is guaranteed to pass the second time around, regardless of races

Reply:
Interestingly enough, I can't repro the failures with the `BashRunner`, so I'm not going to add the overhead of a new enlistment per test case for now. It must be that having to spawn a new bash process per request slows things down enough to avoid hitting these races, because the `SystemIORunner` fails completely reliably and the `BashRunner` passes completely reliably, on my MacBook Pro.